adapters/melozen/melozen.go

```diff
@@ -0,0 +1,201 @@
+package melozen
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpointTemplate *template.Template
+}
+
+// Builder builds a new instance of the {bidder} adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpointTemplate: template,
+	}
+
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errors []error
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	requestCopy := *request
+	for _, imp := range request.Imp {
+		// Extract Sharethrough Params
+		var strImpExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &strImpExt); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		var strImpParams openrtb_ext.ImpExtMeloZen
+		if err := json.Unmarshal(strImpExt.Bidder, &strImpParams); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		if strImpParams.PubId == "" {
+			return nil, []error{&errortypes.BadInput{
+				Message: "The publisher ID must not be empty",
+			}}
+		}
+		url, err := a.buildEndpointURL(strImpParams.PubId)
```

Comment:
`buildEndpointURL` is simple function parsing endpoint template. This code can be inlined here

```
url, err := macros.ResolveMacros(a.endpointTemplate, macros.EndpointTemplateParams{PublisherID: strImpParams.PubId})

```

Reply:
Thank you for the suggestion to inline the macros.ResolveMacros call. I agree that simplifying the code by removing the buildEndpointURL function makes the overall process more straightforward and reduces unnecessary abstraction. I have updated the code accordingly. Please let me know if there's anything else that needs attention!